packages/discord.js-utilities/src/lib/MessagePrompter.ts

```diff
@@ -0,0 +1,402 @@
+					response,
+					options: this.options,
+					appliedMessage: this.appliedMessage,
+					message: this.message
+			  }
+			: response;
+	}
+
+	private async collectReactions(
+		channel: TextChannel | NewsChannel | DMChannel,
+		authorOrFilter: User | CollectorFilter,
+		reactions: string[] | EmojiIdentifierResolvable[],
+		timeout: number
+	): Promise<IMessagePrompterExplicitReturn> {
+		this.appliedMessage = await channel.send(this.message);
+
+		const collector = this.appliedMessage.createReactionCollector(this.createReactionPromptFilter(reactions, authorOrFilter), {
+			max: 1,
+			time: timeout
+		});
+
+		let resolved = false;
+		const collected: Promise<MessageReaction> = new Promise((resolve) =>
+			collector.on('collect', (r) => {
+				resolve(r);
+				resolved = true;
+				collector.stop();
+			})
+		);
+
+		for (const reaction of reactions) {
+			if (resolved) break;
+
+			await this.appliedMessage.react(reaction);
+		}
+
+		const firstReaction = await collected;
+		const emoji = firstReaction?.emoji;
+
+		const reaction = reactions.find((r) => (emoji?.id ?? emoji?.name) === r);
+
+		return {
+			emoji,
+			reaction,
+			options: this.options,
+			appliedMessage: this.appliedMessage,
+			message: this.message
+		};
+	}
+
+	/**
+	 * Creates a filter for the collector to filter on
+	 * @return The filter for awaitReactions function
+	 */
+	private createReactionPromptFilter(reactions: string[] | EmojiIdentifierResolvable[], authorOrFilter: User | CollectorFilter): CollectorFilter {
+		return async (reaction: MessageReaction, user: User) =>
+			reactions.includes(reaction.emoji.id ?? reaction.emoji.name) &&
+			(typeof authorOrFilter === 'function' ? await authorOrFilter(reaction, user) : user.id === authorOrFilter.id) &&
+			!user.bot;
+	}
+
+	/**
+	 * Creates a filter for the collector to filter on
+	 * @return The filter for awaitMessages function
+	 */
+	private createMessagePromptFilter(authorOrFilter: User | CollectorFilter): CollectorFilter {
+		return async (message: Message) =>
+			(typeof authorOrFilter === 'function' ? await authorOrFilter(message) : message.author.id === authorOrFilter.id) && !message.author.bot;
+	}
+
+	/**
+	 * The default options of this handler per type.
+	 * Default is always used, and overriden later.
+	 */
+	public static defaultOptions: Map<IMessagePrompterOptionsType, Partial<IMessagePrompterOptions>> = new Map([
+		[
+			'confirm',
+			{
+				type: 'confirm',
+				reactions: {
+					confirm: '🇾',
+					cancel: '🇳'
+				}
+			}
+		],
+		[
+			'number',
+			{
+				type: 'number',
+				reactions: {
+					start: 0,
+					end: 10
+				}
+			}
+		],
+		[
+			'reaction',
+			{
+				type: 'reaction'
+			}
+		],
+		[
+			'message',
+			{
+				type: 'message'
+			}
+		]
+	]);
+}
+
+/**
+ * A type to extend multiple discord types and simplify usage in [[MessagePrompter]]
+ */
+export type IMessagePrompterMessage = APIMessageContentResolvable | (MessageOptions & { split?: false }) | MessageAdditions;
```

Comment:
`I`nterfaces are prefixed with `I`, not `type`s. The idea is that you can make a class implementing an interface prefixed with `I`, while you usually can't for types.

Reply:
Same applies for all the other types.

Reply:
Want it to be prefixed with `T` or nothing at all?

Reply:
Nothing, not a soul to be seen!